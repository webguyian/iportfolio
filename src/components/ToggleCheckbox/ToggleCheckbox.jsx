import React from 'react';
import { bool } from 'prop-types';
import cx from 'classnames';

import Text from 'components/Text/Text';

const ToggleCheckbox = props => {
  const { checked, disabled } = props;
  const baseClass = 'ui-toggle-checkbox';
  const disabledClass = disabled && `${baseClass}--disabled`;

  return (
    <label className={cx(baseClass, disabledClass)}>
      <input type="checkbox" defaultChecked={checked} disabled={disabled} />
      <i></i>
      <Text type="accessible">{checked ? 'On' : 'Off'}</Text>
    </label>
  );
};

ToggleCheckbox.propTypes = {
  checked: bool,
  disabled: bool
};

ToggleCheckbox.defaultProps = {
  checked: false,
  disabled: false
};

export default ToggleCheckbox;
