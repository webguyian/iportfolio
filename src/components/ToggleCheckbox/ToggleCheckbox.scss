.ui-toggle-checkbox {
  display: inline-block;
  cursor: pointer;

  &--disabled {
    cursor: default;
  }

  > input {
    @include accessibly-hidden;

    &:checked {
      & + i {
        background: $green-bright;
      }

      & + i::after {
        transform: translateX(20px);
      }
    }

    &:disabled {
      & + i {
        opacity: 0.5;
      }
    }

    &:focus {
      & + i {
        outline: 2px solid $blue;
      }
    }
  }

  > i {
    display: inline-block;
    position: relative;
    width: 40px;
    height: 20px;
    padding: 2px;
    transition: 0.25s 0.1s;
    border-radius: 13px;
    background-color: $gray-medium;
    vertical-align: middle;
    box-sizing: initial;

    &::after {
      display: block;
      position: absolute;
      left: 2px;
      width: 20px;
      height: 20px;
      transition: 0.25s;
      border-radius: $border-radius-round;
      background: $white;
      content: " ";
    }
  }
}
