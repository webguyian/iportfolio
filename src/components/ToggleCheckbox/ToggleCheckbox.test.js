import React from 'react';
import { render } from '@testing-library/react';

import ToggleCheckbox from './ToggleCheckbox';

describe('<ToggleCheckbox />', () => {
  const props = {
    checked: true
  };

  it('renders correctly', () => {
    const { asFragment } = render(<ToggleCheckbox {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly unchecked', () => {
    const { asFragment } = render(<ToggleCheckbox />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly disabled', () => {
    const { asFragment } = render(<ToggleCheckbox disabled />);

    expect(asFragment()).toMatchSnapshot();
  });
});
