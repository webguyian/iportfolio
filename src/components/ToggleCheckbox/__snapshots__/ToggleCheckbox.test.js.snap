// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleCheckbox /> renders correctly 1`] = `
<DocumentFragment>
  <label
    class="ui-toggle-checkbox"
  >
    <input
      checked=""
      type="checkbox"
    />
    <i />
    <span
      class="ui-text ui-text--accessible"
    >
      On
    </span>
  </label>
</DocumentFragment>
`;

exports[`<ToggleCheckbox /> renders correctly disabled 1`] = `
<DocumentFragment>
  <label
    class="ui-toggle-checkbox ui-toggle-checkbox--disabled"
  >
    <input
      disabled=""
      type="checkbox"
    />
    <i />
    <span
      class="ui-text ui-text--accessible"
    >
      Off
    </span>
  </label>
</DocumentFragment>
`;

exports[`<ToggleCheckbox /> renders correctly unchecked 1`] = `
<DocumentFragment>
  <label
    class="ui-toggle-checkbox"
  >
    <input
      type="checkbox"
    />
    <i />
    <span
      class="ui-text ui-text--accessible"
    >
      Off
    </span>
  </label>
</DocumentFragment>
`;
